### PR TITLE
[k8s] Fix generate_kubeconfig.sh when ingress is not used

### PR DESCRIPTION
**Pull request by romilbhardwaj**
generate_kubeconfig.sh would fail when a cluster doesn't have ingress:
```
Error from server (NotFound): error when creating "STDIN": namespaces "ingress-nginx" not found
Error from server (NotFound): error when creating "STDIN": namespaces "ingress-nginx" not found
```
This PR makes it optional, and any errors due to ingress not being present are ignored.

### CODE REVIEW

File: sky/utils/kubernetes/generate_kubeconfig.sh [hunk 1]
```diff
@@ -169,6 +169,9 @@ roleRef:
   kind: Role
   name: skypilot-system-service-account-role
   apiGroup: rbac.authorization.k8s.io
+EOF
+# Apply optional ingress-related roles, but don't make the script fail if it fails
+kubectl apply -f - <<EOF || true
```
Comment (Michaelvll) [inline sky/utils/kubernetes/generate_kubeconfig.sh:174]: Should we `echo` some message when it fails?

File: sky/utils/kubernetes/generate_kubeconfig.sh [hunk 2]
```diff
@@ -169,6 +169,9 @@ roleRef:
...
 ---
```
Comment (Michaelvll) [inline sky/utils/kubernetes/generate_kubeconfig.sh:175]: Do we need this `---` here?

File: sky/utils/kubernetes/generate_kubeconfig.sh (hunk 1 above)
Comment (romilbhardwaj) [reply sky/utils/kubernetes/generate_kubeconfig.sh]: thanks, added!

File: sky/utils/kubernetes/generate_kubeconfig.sh (hunk 2 above)
Comment (romilbhardwaj) [reply sky/utils/kubernetes/generate_kubeconfig.sh]: good catch, removed now.

**Review by Michaelvll — APPROVED** [review]

**Pull request merged**